### ISSUE TITLE
False positive: `module is private` in tests

### ISSUE THREAD

**Issue by qiujiangkun** [intellij-rust/intellij-rust#8383]
## Environment

* **IntelliJ Rust plugin version:** 0.4.162.4321-213
* **Rust toolchain version:** 1.59.0-nightly (e012a191d 2022-01-06) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2021.3.2 (CL-213.6461.75)
* **Operating system:** Linux 5.15.12-1-manjaro
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description
![image](https://user-images.githubusercontent.com/33482468/148763860-9eefb234-86ff-4ff3-a620-561480537d70.png)
Module `test_util` is private [E0603]

## Steps to reproduce

tests/utils/mod.rs
```rust
pub fn foo() {}
```

tests/test_foo.rs
```rust
mod utils;
use utils::foo;
```
